legacy/src/main/java/com/amazon/opendistroforelasticsearch/sql/legacy/plugin/RestSQLQueryAction.java

```diff
@@ -179,4 +197,17 @@ private void sendResponse(RestChannel channel, RestStatus status, String content
         status, "application/json; charset=UTF-8", content));
   }
 
+  private static void logAndPublishMetrics(Exception e) {
```

Comment:
I'm thinking should we publish server error metrics always in this method? Because at this point we've passed parsing, analyzing and planning, I assume any exception occurred should be treated as server error.

But this seems not true from the `SELECT DATE('0')` example in PR description. The semantic error can also be thrown here?

Reply:
+1, after analyer/parser, all the execption should be treat as SYS error. The example like is DATE('0') may throw through runtim execption, but it is hard to know it is because of CUS error or data issue. I think it should be fine to treat it as SYS error.